photon-ml/src/main/scala/com/linkedin/photon/ml/avro/data/DataProcessingUtils.scala

```diff
@@ -117,9 +117,24 @@ object DataProcessingUtils {
     } else {
       1.0
     }
+
+    val metaMap = Utils.getMapAvro(record, AvroFieldNames.META_DATA_MAP)
     val ids = randomEffectIdSet
       .map { randomEffectId =>
-        (randomEffectId, Utils.getStringAvro(record, randomEffectId))
+        val idValue = Utils.getStringAvro(record, randomEffectId, isNullOK = true)
+        val finalIdValue = if (idValue == null) {
+          val mapIdValue = metaMap.get(randomEffectId)
+          if (mapIdValue == null) {
+            throw new IllegalArgumentException(s"Cannot find id in either record" +
+              s"field: ${randomEffectId} or in metadataMap with key: #${randomEffectId}")
+          }
```

Comment:
Yes I know we call ```toString``` in the end, but I was wondering don't we need to return anything after the ```if (mapIdValue == null) {...}``` branch, or did I miss anything?